models/migrations/v1_23/v300.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_23 //nolint
+
+import "xorm.io/xorm"
+
+func AddForcePushBranchProtection(x *xorm.Engine) error {
+	type ProtectedBranch struct {
+		WhitelistUserIDs             []int64 `json:"whitelist_user_ids" xorm:"JSON TEXT"`
+		WhitelistTeamIDs             []int64 `json:"whitelist_team_ids" xorm:"JSON TEXT"`
+		EnableForcePushAllowlist     bool
+		ForcePushAllowlistUserIDs    []int64 `json:"force_push_allowlist_user_ids" xorm:"JSON TEXT"`
+		ForcePushAllowlistTeamIDs    []int64 `json:"force_push_allowlist_team_ids" xorm:"JSON TEXT"`
+		ForcePushAllowlistDeployKeys bool    `xorm:"NOT NULL DEFAULT false"`
+		MergeWhitelistUserIDs        []int64 `json:"merge_whitelist_user_ids" xorm:"JSON TEXT"`
+		MergeWhitelistTeamIDs        []int64 `json:"merge_whitelist_team_ids" xorm:"JSON TEXT"`
+		ApprovalsWhitelistUserIDs    []int64 `json:"approvals_whitelist_user_ids" xorm:"JSON TEXT"`
+		ApprovalsWhitelistTeamIDs    []int64 `json:"approvals_whitelist_team_ids" xorm:"JSON TEXT"`
```

Comment:
The migration struct should be synchronized with the model struct.

Reply:
Ahh, good catch.

Updated in https://github.com/go-gitea/gitea/pull/28086/commits/410eb5980c994167fda200bb2bab5060ef3fe9d4